File: python/design-patterns/bridge/concrete-devices/concrete_devices.py
class Tv(Device):
    def __init__(self):
        self._on = False
        self._volume = 30
    def is_enabled(self): return self._on
    def enable(self): self._on = True
    def disable(self): self._on = False
    def get_volume(self): return self._volume
    def set_volume(self, p): self._volume = max(0, min(100, p))

class Radio(Device):
    def __init__(self):
        self._on = False
        self._volume = 15
    def is_enabled(self): return self._on
    def enable(self): self._on = True
    def disable(self): self._on = False
    def get_volume(self): return self._volume
    def set_volume(self, p): self._volume = max(0, min(100, p))

File: cpp/design-patterns/bridge/concrete-devices/concrete_devices.cpp
class Tv : public Device {
    bool on = false;
    int volume = 30;
public:
    bool isEnabled() override { return on; }
    void enable() override { on = true; }
    void disable() override { on = false; }
    int getVolume() override { return volume; }
    void setVolume(int p) override { volume = p; }
};